Source/Array.swift

```diff
@@ -1,24 +1,79 @@
+/**
+    map a function over an array of values
+
+    This will return a new array resulting from the transformation function beind applied to each value in the array
+
+    :param: f A transformation function from type T to type U
+    :param: a An array of type [T]
+
+    :returns: An array of type [U]
+*/
 public func <^><T, U>(f: T -> U, a: [T]) -> [U] {
     return a.map(f)
 }
 
+/**
+    apply an array of functions to an array of values
+
+    This will return a new array resulting from the matrix of each function being applied to each value in the array
```

Comment:
My first take is:

    This will return a new array resulting by appling each function to each value in the array

But maybe we can just use a different word. Maybe: sequence, mapping, list, vector, array?

... maybe matrix is fine.